### ISSUE TITLE
old translations reappearing

### ISSUE THREAD

**Issue by developwithvelocity** [svenfuchs/i18n-active_record#13]
Hi,

We use store_translations to store a new translation and then later use I18n.t to retrieve the translation. This works pretty well, but sometimes a previously stored translation will mysteriously re-appear on our web site after a few page refreshes.  We've checked the database and it has the correct value.  We've also cleared the browser cache, and and that doesn't help. 

Has any one else seen this problem?

Thanks,
Ruby Duo

**Comment by xinspire** [comment]
So I'm running two app instances on AWS under a load balancer, and even though I call `I18n.backend.reload!` after `create`, `update`, and `destroy`, I think it only applies to the instance that's handling that request.  When I refresh the page in the browser, occasionally, I still get the old translation.  Any ideas on how to workaround this?

**Comment by ChrisHampel** [comment]
What about adding a controller function which will compare the current max(:updated_at) of the translation table to a stored value on that server to check if the translation has been updated since this server last updated it's translation. Then you could just do:
before_action :update_i18n  ,  :if =>: i18n_changed

**Comment by xinspire** [comment]
@ChrisHampel - Doesn't that negate the advantages of caching if you have to hit the DB to check the translation updated_at for every request?

**Comment by xinspire** [comment]
Okay, I ended up implementing the "check" using Redis instead of doing a SQL query.  Both instances will now reload via `I18n.backend.reload!` on the first request after a translation is updated.

However, as I was testing, I noticed that the old data occasionally still showed.  When I took out `I18n::Backend::ActiveRecord.send(:include, I18n::Backend::Memoize)` from the initializer, the "old translation" problem was solved, but now I have a caching problem.

**Comment by xinspire** [comment]
UPDATE: Seems like this worked:
`I18n::Backend::ActiveRecord.send(:include, I18n::Backend::Cache)`
`I18n.cache_store = ActiveSupport::Cache.lookup_store(:memory_store)`

**Comment by fmh** [comment]
same issue !

**Comment by timsly** [comment]
There is a possible solution here plus related fix in #85 
So I am closing this one, but feel free to reopen it if issue is still there

**Comment by gmonein** [comment]
If you use unicorn you need to reload i18n in all instance,
So you need to create a FileUpdateChecker for you app instance, and check on every request if files have changed

```
i18n_reloader = ActiveSupport::FileUpdateChecker.new(Dir["#{Rails.root.to_path}/config/locales/*yml"]) do
  I18n.backend.reload!
end

ApplicationController.before_action do
  i18n_reloader.execute_if_updated
end
```
